### PR TITLE
doc(routing): add a routing guide

### PR DESCRIPTION
**Pull request by bobylito**
Adds a guide for the new routing feature

Obviously this is not ready, yet. Happy to have ideas on the kind of content you want here :)

### CODE REVIEW

File: docgen/src/guides/sync-url.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+And those two methods should comply to the following rule: for any `uiState`, `uiState` must be equal in
+value to `routeToState(stateToRoute(uiState))`. This means that any transformation applied by `stateToRoute`
+can be cancelled by `routeToState`.
+
+Let's take an example:
+
+```javascript
+// state mapping that extracts brand from refinementList
+// and put the values in a string
```
Comment (Haroenv) [inline docgen/src/guides/sync-url.md]: is this missing the actual code?

File: docgen/src/guides/sync-url.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+Sometimes the URL is not the best way to store the search. The routing API
+is flexible and let you work with other outputs.
+
+A router has multiple interesting properties:
+ - writes and reads from a storage
+ - can be triggered own its own (pressing the back button in a browser)
+ - can optionaly generated an ID / URL for the current state
+
+Let's see an example in which we implement a router for the localStorage.
```
Comment (Haroenv) [inline docgen/src/guides/sync-url.md]: is this missing the actual example?

File: docgen/src/guides/sync-url.md (hunk 1 above)
Comment (bobylito) [reply docgen/src/guides/sync-url.md]: it's not ready yet :) Don't bother review the content. This is a placeholder

**Comment by vvo** [comment]
Moved to https://github.com/algolia/instantsearch.js/pull/2879

**Pull request closed without merging**